### ISSUE TITLE
why add  (i*channels)  in  dot product operation?   why have two weight?

### ISSUE THREAD

**Issue by htt123tth** [kevinlin311tw/cvpr16-deepbit#20]
Hello, can help me explain this loss code, I do not understand, thank you very much！For example, why add  (i*channels)  in  dot product operation?   why have two weight?

for (int i = 0; i < bottom[0]->num(); ++i) {
    dist_sq_.mutable_cpu_data()[i] = caffe_cpu_dot(channels,
        diff_.cpu_data() + (i*channels), diff_.cpu_data() + (i*channels));

    loss += (rotation_weight2[i]*(rotation_weight1[i]*dist_sq_.cpu_data()[i]));
  }
